rskj-core/src/main/java/co/rsk/net/sync/PeersInformation.java

```diff
@@ -199,7 +199,7 @@ public List<Peer> getBestPeerCandidates() {
     @Override
     public List<Peer> getBestSnapPeerCandidates() {
         return getBestCandidatesStream()
-                .filter(this::isSnapPeerCandidate)
+                .filter(entry -> isSnapPeerCandidate(entry) && entry.getKey().isSnapCapable())
```

Comment:
Got it! I’ve updated the methods to use an OR condition so that both discovered snap capable and config-listed peers are considered. Thanks for the feedback!

Reply:
I think we shouldn't check the list of snap candidates here, what it is important is the capability. 
I mean if a node is in the list but it is not snapCapable, then we are going to start the communication but it won't attend the snap status request.

The list should be used to add these nodes to the node list if the snap sync is enabled.

Reply:
I see your point about prioritizing snap capability over the candidate list, but that makes me wonder—what’s the purpose of having the snap candidate list in the config then? If we’re already filtering based on capability, should we just focus on all snap-capable peers, whether or not they’re in the list?

It feels like the list might not be adding much value if we’re not using it to narrow down the best snap peers. Should we be considering a different approach to how we use the list?

Reply:
This was already discussed before, the reason is to have a list of nodes pre-configured for snap sync if the feature is enabled, as same is happening with the peer discovery list. 
Apart from that I don't see any positive thing to send snap messages to a client, knowing forehand that it is not snap capable. It will only delay the sync and spread unnecessary packages in the network.